runtime/doc/lua.txt

```diff
@@ -360,32 +360,36 @@ arguments separated by " " (space) instead of "\t" (tab).
 luaeval()                                                 *lua-eval* *luaeval()*
 
 The (dual) equivalent of "vim.eval" for passing Lua values to Nvim is
-"luaeval". "luaeval" takes an expression string and an optional argument used
-for _A inside expression and returns the result of the expression. It is
-semantically equivalent in Lua to: >
+"luaeval". "luaeval" takes an expression string and a second optional argument
+to be used by the expression string itself to manipulate its own arguments.
+The expression is then computed and returns the final result.
+
+The magic global "_A" contains the second argument to luaeval().
+
+Example: >
+    :echo luaeval('_A[1] + _A[2]', [40, 2])
+    42
+    :echo luaeval('string.match(_A, "[a-z]+")', 'XYXfoo123')
+    foo
+<
+
+"luaeval" is semantically equivalent in Lua to: >
 
     local chunkheader = "local _A = select(1, ...) return "
     function luaeval (expstr, arg)
         local chunk = assert(loadstring(chunkheader .. expstr, "luaeval"))
         return chunk(arg) -- return typval
     end
 <
+
 Lua nils, numbers, strings, tables and booleans are converted to their
 respective Vimscript types. If a Lua string contains a NUL byte, it will be
 converted to a |Blob|. Conversion of other Lua types is an error.
 
-The magic global "_A" contains the second argument to luaeval().
-
-Example: >
-    :echo luaeval('_A[1] + _A[2]', [40, 2])
-    42
-    :echo luaeval('string.match(_A, "[a-z]+")', 'XYXfoo123')
-    foo
-<
```

Comment:
I moved this snippet up because I spent some time trying to understand what `A_` meant without seeing the example first.